spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/reactor/ReactorSleuth.java

```diff
@@ -115,10 +115,19 @@ private ReactorSleuth() {
 				return sub; // no need to scope a null parent
 			}
 
+			// Handle scenarios such as Mono.defer
+			if (sub instanceof ScopePassingSpanSubscriber) {
+				ScopePassingSpanSubscriber<?> scopePassing = (ScopePassingSpanSubscriber) sub;
+				if (scopePassing.parent.equals(parent)) {
+					return sub; // don't double-wrap
```

Comment:
the way I figured out defer did this, was I changed this line to throw an exception, then ran all tests. Defer ended up one of the patterns in the stack, though there are likely others.

Reply:
nice idea.  
Unfortunately it works only if there is only one lift function in `onEachOperator` hook.

Reply:
to clarify, the optimization is only _guaranteed_ to work if there is only one lift function. having several sleuth lift functions chained together would also work, or if sleuth is the last lift function in the chain. this is good enough coverage to make it worth applying the optim, I guess